src/software/amazon/ion/Timestamp.java

```diff
@@ -276,14 +277,33 @@ private void apply_offset(int offset)
     @SuppressWarnings("deprecation")
     private void set_fields_from_millis(long millis)
     {
-        if(millis < MINIMUM_ALLOWED_TIMESTAMP_IN_MILLIS){
+        if(millis < MINIMUM_ALLOWED_TIMESTAMP_IN_MILLIS) {
             throw new IllegalArgumentException("year is less than 1");
         }
 
         Date date = new Date(millis);
 
-        // These fields are in the system timezone!
-        this._year    = checkAndCastYear(date.getYear() + 1900);
+        // java.util.Date has a Date(long) constructor that expects an epoch time in milliseconds. Date's getYear(),
```

Comment:
I think we can simplify this comment;  what do you think about the following?

```
// https://github.com/amzn/ion-java/issues/160
// The java.util.Date(long) constructor expects an epoch time in milliseconds, and getYear(), getMonth(),
// getHour() on the resulting Date are supposed to return values adjusted to the default timezone.
// In Pacific Standard Time (offset -08:00), for a Date constructed with an epoch time equivalent to
// 0001-01-01T00:00:00.000Z, the Date.get*() methods should return values for
// 0000-12-31T16:00:00.000Z;  however, Date.getYear() incorrectly returns 1 in this scenario.
// The following if/else block compensates for this bug:
```

Reply:
It actually returns -1899 (the year -1900) for some reason unfathomable to me.

I've modified this slightly:
```
// https://github.com/amzn/ion-java/issues/160
// The java.util.Date(long) constructor expects an epoch time in milliseconds, and getYear(), getMonth(),
// getHour() on the resulting Date are supposed to return values adjusted to the default timezone.
// In Pacific Standard Time (offset -08:00), for a Date constructed with an epoch time equivalent to
// 0001-01-01T00:00:00.000Z, the Date.get*() methods should return values for
// 0000-12-31T16:00:00.000Z;  however, Date.getYear() incorrectly returns a value for year 1 (-1899)
// in this scenario. The following if/else block compensates for this bug:
```